src/scripts/main.js

```diff
@@ -1,3 +1,221 @@
 'use strict';
 
-// write your code here
+const gameSize = 4;
+const startBtn = document.querySelector('.button');
+const startMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+const scoreField = document.querySelector('.game-score');
+let score = 0;
+let direction;
+let gameSquare = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+
+];
+
+startBtn.addEventListener('click', () => {
+  newGame();
+
+  if (startBtn.innerText === 'Start') {
+    startBtn.innerText = 'Restart';
+    startBtn.classList.replace('start', 'restart');
+  }
+
+  startMessage.classList.add('hidden');
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+});
+
+function generateNewNumber() {
+  const randomRow = Math.floor(Math.random() * gameSize);
+  const randomCell = Math.floor(Math.random() * gameSize);
+
+  if (gameSquare[randomRow][randomCell] === 0) {
+    gameSquare[randomRow][randomCell] = Math.random() > 0.1 ? 2 : 4;
+  } else {
+    generateNewNumber();
+  }
+}
+
+function fillGameField() {
+  const gameBoard = document.querySelector('.game-field');
+  const rows = [...gameBoard.rows];
+  const gameBoardCells = [];
+
+  for (const a of rows) {
+    gameBoardCells.push([...a.cells]);
+  }
+
+  for (let row = 0; row < gameSize; row++) {
+    for (let column = 0; column < gameSize; column++) {
+      const gameValue = gameBoardCells[row][column];
+      const arrayValue = gameSquare[row][column];
+
+      gameValue.innerText = arrayValue === 0 ? '' : arrayValue;
+      gameValue.classList.value = '';
+
+      gameValue.classList.add('field-cell',
+        `field-cell--${gameValue.innerText}`);
+    }
+  }
+}
+
+function newGame() {
+  gameSquare = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
```

Comment:
use a variable with default state

Reply:
move `[
    [0, 0, 0, 0],
    [0, 0, 0, 0],
    [0, 0, 0, 0],
    [0, 0, 0, 0],
  ] ` this part to a variable and use it when you need